src/lib/eliom_cscache.eliom

```diff
@@ -47,6 +47,25 @@ let%server find cache get_data id =
     do_cache_raw cache id th;
     th
 
+let cache_list cache get_data ids =
+  let main_thread, wakeup = Lwt.task () in
+  let tbl = Hashtbl.create 10 in
+  let thread id =
+    let%lwt () = main_thread in
+    Lwt.return @@ Hashtbl.find tbl id
+  in
+  let enqueue id =
+    try let _ = Hashtbl.find (Eliom_shared.Value.local cache ()) id in false
+    with Not_found -> let () = do_cache_raw cache id (thread id) in true
+  in
+  try%lwt
+    let not_cached = List.filter enqueue ids in
+    let%lwt data = get_data not_cached in
+    List.iter (fun (id,v) -> Hashtbl.add tbl id v) data;
+    ignore [%client (List.iter (fun (id,v) -> do_cache ~%cache id v) ~%data : unit)];
+    Lwt.return @@ Lwt.wakeup wakeup ()
+  with e -> Lwt.return @@ Lwt.wakeup_exn wakeup e
```

Comment:
That should probably be:
```ocaml
Lwt.wakeup_exn wakeup e;
Lwt.fail e
```
In our case, we probably want the RPC to fail if the database query fails, rather than returning a result that triggers a lot of additional RPCs to fetch the missing data.